src/main/java/org/httpserver/core/ServerListenerThread.java

```diff
@@ -0,0 +1,45 @@
+package org.httpserver.core;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.Socket;
+import java.net.ServerSocket;
+
+public class ServerListenerThread extends Thread {
+    private final static Logger LOGGER = LoggerFactory.getLogger(ServerListenerThread.class);
+    private int port;
+    private String webroot;
+    private ServerSocket serverSocket;
+    public ServerListenerThread(int root, String webroot) throws IOException {
+        this.port = root;
+        this.webroot = webroot;
+        this.serverSocket = new ServerSocket(this.port);
+    }
+
+    @Override
+    public void run() {
+        try {
+            while (serverSocket.isBound() && !serverSocket.isClosed() ) {
```

Comment:
Based on the logic written here, I assume the next:
- we start processing first request, at this time socket is bound and open.
- in `finally` block we close the socket.
- after that, when we will be processing the next request, this check will not pass, since the socket has been closed earlier.